common/src/main/java/org/apache/nemo/common/DirectByteBufferOutputStream.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.common;
+
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.LinkedList;
+
+/**
+ * This class is a customized output stream implementation backed by
+ * {@link ByteBuffer}, which utilizes off heap memory when writing the data.
+ * Memory is allocated when needed by the specified {@code pageSize}.
+ */
+public final class DirectByteBufferOutputStream extends OutputStream {
+
+  private final LinkedList<ByteBuffer> dataList = new LinkedList<>();
+  private static int pageSize = 4096;
```

Comment:
By following the Nemo coding conventions, this variable should be `PAGE_SIZE`, instead of `pageSize`, I guess :)

Reply:
This variable is not `final`, so I guess `pageSize` matches the static variable naming convention. Thanks :)

Reply:
As `pageSize` is modified inside the instance, it should not be `static`. I suggest to define `DEFAULT_PAGE_SIZE` as a static variable instead of `pageSize`.